src/prototype_room_my.js

```diff
@@ -54,7 +54,16 @@ Room.prototype.getLinkStorage = function() {
       return link;
     }
   }
-  const linkPos = this.memory.position.structure.link[0];
+
+  let linkPos = null;
```

Comment:
In the room setup `prototype_room_basebuilder` the first link is placed next to the filler, also the link transfer logic always transfer the energy to the first link.

If this does not work out, we should check what happened on the initial room setup.

If this helps for your rooms, you could also just add a log output in case the first link is not the one desired, something like: 'First link is not the best for filler, please manually check room setup'. So that someone ./ me can look into it, in case it happens for me, too.

Reply:
I've checked the code currently the `updatePosition` from `prototype_room_init.js` [Link](https://github.com/TooAngel/screeps/blob/e99584fddd04d97baa7eae1f089a50a1b5d7363e/src/prototype_room_init.js#L128-L135) firstly adds the links for the sources inside `initSetSources()` afterwards the storage link is generated inside `setFillerArea()`.
I think this should be fixed and my check should be included for a log statement.

Reply:
I just checked my rooms and all rooms have this wrong link ordering.

Reply:
Ok, thx, that 'bug' was introduced ... by me :-) I opened a new issue for it #582 

So if you could include a logline here, that would be cool. Let's fix it in another PR.

Reply:
Okay I will add the logline in another spot and remove the new code. 👍 
Maybe it would be good to add an option `config.debug.checkbugs` so that this check is not executed always?

Reply:
Yeah, sounds good